NullPointerException in DatabaseChangeLog.ranChangeSetExists when storedChangeLog is null (MongoDB + logicalFilePath)

### Search first

- [x] I searched and no similar issues were found

### Description

Liquibase throws a NullPointerException when running migrations against MongoDB if the stored changelog entry does not contain `storedChangeLog`.

The error happens when changelog files were renamed and `logicalFilePath` was introduced to prevent Liquibase from re-running migrations.

When Liquibase processes changelogs using `includeAll`, it calls `DatabaseChangeLog.ranChangeSetExists()` and attempts to compare:

ranChangeSet.getStoredChangeLog().equals(...)

However `storedChangeLog` is `null`, which causes a NullPointerException.

**Stacktrace:**
Caused by: java.lang.NullPointerException:
Cannot invoke "String.equals(Object)" because the return value of
"liquibase.changelog.RanChangeSet.getStoredChangeLog()" is null

at liquibase.changelog.DatabaseChangeLog.ranChangeSetExists(DatabaseChangeLog.java:1187)



This happens when the DATABASECHANGELOG collection already contains entries created by an earlier Liquibase version where `storedChangeLog` was not stored.

Liquibase should handle this case safely instead of throwing an exception.

### Steps To Reproduce

1. Run Liquibase MongoDB migrations using includeAll
   Root changelog loads all files from the changelog directory.
{
  "databaseChangeLog": [
    {
      "includeAll": {
        "path": "db/changelogs"
      }
    }
  ]
}
One of the files inside the directory:
  db/changelogs/changelog-1.0.json


1. Run Liquibase MongoDB migrations using includeAll
   Root changelog loads all files from the changelog directory.
{
  "databaseChangeLog": [
    {
      "includeAll": {
        "path": "db/changelogs"
      }
    }
  ]
}

One of the files inside the directory:
db/changelogs/changelog-1.0.json

3. Rename the changelog file

The file is renamed:
changelog-1.0.json → changelog-001.0.json

4. Add logicalFilePath to keep the original Liquibase identity

To prevent Liquibase from re-running the migration, logicalFilePath is added to the renamed file:
{
  "databaseChangeLog": [
    {
      "logicalFilePath": "db/changelogs/changelog-1.0.json"
    }
  ]
}

5. Run Liquibase again

Liquibase loads changelog files using includeAll.

During the process of checking if a changeset already ran, Liquibase throws:

NullPointerException: ranChangeSet.getStoredChangeLog() is null

This occurs while Liquibase compares the stored changelog entry with the current changelog file.



### Expected/Desired Behavior

Liquibase should not throw a NullPointerException when `storedChangeLog` is null.

Instead it should handle the comparison safely, for example using a null-safe comparison:

Objects.equals(ranChangeSet.getStoredChangeLog(), changeLog)

This would allow Liquibase to correctly detect existing changesets even if older DATABASECHANGELOG entries do not contain `storedChangeLog`.

### Liquibase Version

4.33.0

### Database Vendor & Version

MongoDB 7.x

### Liquibase Integration

Quarkus Liquibase MongoDB Extension (Gradle)

### Liquibase Extensions

liquibase-mongodb

### OS and/or Infrastructure Type/Provider

Windows 10 (development)

### Additional Context

The issue appeared after upgrading Liquibase versions.

Older DATABASECHANGELOG records created by previous Liquibase versions do not contain `storedChangeLog`.

When Liquibase processes changelogs via `includeAll`, it attempts to compare the stored changelog with the current changelog and crashes due to the null value.

This situation commonly happens when:
- changelog files are renamed
- logicalFilePath is used to preserve migration history
- DATABASECHANGELOG entries were created by older Liquibase versions

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/changelog/RanChangeSet.java
+++ b/liquibase-standard/src/main/java/liquibase/changelog/RanChangeSet.java
@@ -86,6 +86,13 @@ public class RanChangeSet {
         return (Date) dateExecuted.clone();
     }
 
+    public String getStoredChangeLog() {
+        if (storedChangeLog == null) {
+            return changeLog;
+        }
+        return storedChangeLog;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) {

--- a/liquibase-standard/src/test/java/liquibase/changelog/RanChangeSetTest.java
+++ b/liquibase-standard/src/test/java/liquibase/changelog/RanChangeSetTest.java
@@ -2,6 +2,7 @@ package liquibase.changelog;
 
 import org.junit.Test;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
@@ -26,5 +27,19 @@ public class RanChangeSetTest {
         RanChangeSet ranChangeSet = new RanChangeSet("db/ran-changeset.log", "1", "author", null, null, null, null, null, null, null, null, null);
         ChangeSet incomingChangeSet = new ChangeSet("1", "author", false, false, "classpath:/db/file.log", null, null, null);
         assertFalse(ranChangeSet.isSameAs(incomingChangeSet));
+    }
+
+    @Test
+    public void stored_change_log_falls_back_to_change_log() {
+        RanChangeSet ranChangeSet = new RanChangeSet("db/file.log", "1", "author", null, null, null, null, null, null, null, null, null);
+
+        assertEquals("db/file.log", ranChangeSet.getStoredChangeLog());
+    }
+
+    @Test
+    public void stored_change_log_uses_explicit_stored_path() {
+        RanChangeSet ranChangeSet = new RanChangeSet("logical/file.log", "1", "author", null, null, null, null, null, null, null, null, null, "physical/file.log");
+
+        assertEquals("physical/file.log", ranChangeSet.getStoredChangeLog());
     }
 }

